cluster/saltbase/salt/kube-master-addons/init.sls

```diff
@@ -37,10 +37,19 @@ master-docker-image-tags:
   file.touch:
     - name: /srv/pillar/docker-images.sls
 
+# Current containervm image by default has both docker and kubelet
+# running. But during cluster creation stage, docker and kubelet
+# could be overwritten completely, or restarted due flag changes.
+# The ordering of salt states for service docker, kubelet and
+# master-addon below is very important to avoid the race between
+# salt restart docker or kubelet and kubelet start master components.
 kube-master-addons:
   service.running:
     - enable: True
     - restart: True
+    - require:
+      - service: docker
+      - service: kubelet
```

Comment:
God, the coffee just isn't kicking this morning. Ignore both comments, I just remembered this is a separate script. So horribly embarrassed.

Okay, I worked out how this helps and now am on the same. I was worried because the manifest files are laid out on disk when Salt runs, and they're laid out in arbitrary order (because Salt). But this service is the only thing doing the `docker load` from the binary tar files, so yes, as @roberthbailey worked out, kubelet won't be able to start them until they're side-loaded. (Slowly waking up.)

Now that I've caught up - this does feel like a noop patch. I'm not sure what guarantees it's trying to provide.